eth/core/genesis.go

```diff
@@ -76,3 +76,10 @@ func UnmarshalGenesisHeader(header *types.Header, gen *Genesis) {
 	gen.Mixhash = header.MixDigest
 	gen.Coinbase = header.Coinbase
 }
+
+// SortableAddresses is a sortable slice of common.Addresses.
+type SortableAddresses []common.Address
+
+func (a SortableAddresses) Len() int           { return len(a) }
```

Comment:
maybe add something that prevents duplicates? so that that doesn't introduce non-determinism / a bad genesis

Reply:
actually will panic earlier as the json struct is being unmarshalled into the golang genesisAccounts map (keyed on addresses)

Reply:
does it have a check for it?

Reply:
geth has no `VerifyGenesis`. It just overwrites it if given multiple accounts with the same address in the map.That behavior makes sense imo

Reply:
just checked "When unmarshaling a JSON object with duplicate keys into a Go map using json.Unmarshal(), the last value for the duplicate key will be used"

Reply:
this could cause non determinism if the order for which sort sorts the two duplicate keys is non deterministic, we should explicitly check for this case and panic with a duplicate key error message

Reply:
there wouldn't be any non-determinism as the json file is read in order, so always the last of the duplicates will be included in the map